libPhoneNumberGeocoding/Metadata/NBGeocoderMetadataHelper.m

```diff
@@ -0,0 +1,119 @@
+//
+//  NBGeocoderMetadataHelper.m
+//  libPhoneNumberiOS
+//
+//  Created by Rastaar Haghi on 6/12/20.
+//  Copyright © 2020 Google LLC. All rights reserved.
+//
+
+#import "NBGeocoderMetadataHelper.h"
+#import "NBPhoneNumber.h"
+
+@implementation NBGeocoderMetadataHelper {
+ @private
+  NSString *_databasePath;
+  sqlite3 *_database;
+  sqlite3_stmt *_selectStatement;
+  NSString *_language;
+  NSNumber *_countryCode;
+  const char *_completePhoneNumber;
+}
+
+NSString *const preparedStatement = @"WITH recursive count(x)"
+                                    @"AS"
+                                    @"( "
+                                    @"SELECT 1 "
+                                    @"UNION ALL "
+                                    @"SELECT x+1 "
+                                    @"FROM   count "
+                                    @"LIMIT  length(?)), tosearch "
+                                    @"AS "
+                                    @"( "
+                                    @"SELECT substr(?, 1, x) AS indata "
+                                    @"FROM   count) "
+                                    @"SELECT   nationalnumber, "
+                                    @"description, "
+                                    @"length(nationalnumber) AS nationalnumberlength "
+                                    @"FROM     geocodingpairs%@ "
+                                    @"WHERE    nationalnumber IN tosearch "
+                                    @"ORDER BY nationalnumberlength DESC "
+                                    @"LIMIT    2";
+static NSString * const kResourceBundleName = @"GeocodingMetadata.bundle";
+
+- (instancetype)initWithCountryCode:(NSNumber *)countryCode withLanguage:(NSString *)languageCode {
+  self = [super init];
+  if (self != nil) {
+    _countryCode = countryCode;
+    _language = languageCode;
+
+    NSBundle *bundle = [NSBundle bundleForClass:self.classForCoder];
+    NSURL *bundleURL = [[bundle resourceURL] URLByAppendingPathComponent:kResourceBundleName];
+    NSString *shortLanguageCode = [[languageCode componentsSeparatedByString:@"-"] firstObject];
+    NSString *databasePath = [NSString stringWithFormat:@"%@%@.db", bundleURL, shortLanguageCode];
+    if (databasePath == nil) {
+      @throw [NSException exceptionWithName:NSInvalidArgumentException
+                                     reason:@"Geocoding Database URL not found"
+                                   userInfo:nil];
+    }
+
+    _databasePath = databasePath;
```

Comment:
do we need to store this?

Reply:
You're right, unnecessary to hold databasePath. Removed.